cmd/api/main.go

```diff
@@ -127,45 +140,106 @@ func responseJSON(w http.ResponseWriter, status int, body any) {
 	}
 }
 
-/* Stores the entry as a new book in the database, if there isn't one with the same name yet. */
-func createBook(w http.ResponseWriter, r *http.Request) {
+func validateBookEntry(w http.ResponseWriter, r *http.Request) (valid bool, bookEntry Book) {
 
-	var newBook Book
-	err := json.NewDecoder(r.Body).Decode(&newBook) //Read the Json body and save the entry to newBook
+	err := json.NewDecoder(r.Body).Decode(&bookEntry) //Read the Json body and save the entry to bookEntry
 	if err != nil {
 		log.Println(err)
 		errR := errResponse{
-			Code:    errResponseCreateBookInvalidJSON.Code,
-			Message: errResponseCreateBookInvalidJSON.Message + err.Error(),
+			Code:    errResponseBookEntryInvalidJSON.Code,
+			Message: errResponseBookEntryInvalidJSON.Message + err.Error(),
 		}
 		responseJSON(w, http.StatusBadRequest, errR)
-		return
+		return false, bookEntry
 	}
 
-	filled := filledFields(newBook) //Verify if all entry fields are filled.
+	filled := filledFields(bookEntry) //Verify if all entry fields are filled.
 	if !filled {
-		responseJSON(w, http.StatusBadRequest, errResponseCreateBookBlankFileds)
+		responseJSON(w, http.StatusBadRequest, errResponseBookEntryBlankFileds)
+		return false, bookEntry
+	}
+
+	return true, bookEntry
+}
+
+/* Verifies with all fields are correctly filled and update the book in the db. */
+func updateBook(w http.ResponseWriter, r *http.Request) {
+	id, err := isolateId(w, r)
+	if err != nil {
+		return
+	}
+
+	//Searching for that ID on database:
+	returnedBook, err := searchById(id)
+	if err != nil {
+		if errors.Is(err, errBookNotFound) {
+			log.Println(err)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		log.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	valid, bookEntry := validateBookEntry(w, r)
+	if !valid {
+		return
+	}
+
+	if returnedBook.Name != bookEntry.Name { //If the update do not change the name of book, we don't need to check sameNameOnBD
+		unique, err := sameNameOnDB(bookEntry) //Verify if the already there is a book with the same name in the database
+		if err != nil {
+			log.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if !unique {
+			responseJSON(w, http.StatusBadRequest, errResponseBookEntryNameConflict)
+			return
+		}
+	}
```

Comment:
I am not feeling good about this, if the user can run into conflicts and we consider `name` as an unique table column maybe we shouldn't let the user change the Books name? I am still not convinced that we should prevent the user from creating books with the same name, we don't use `name` anymore as an identifier anyway.